x-pack/plugins/stack_connectors/public/connector_types/stack/opsgenie/params.tsx

```diff
@@ -0,0 +1,219 @@
+    subActionParams?.message !== undefined;
+
+  return (
+    <>
+      <EuiFormRow
+        data-test-subj="opsgenie-message-row"
+        fullWidth
+        error={errors['subActionParams.message']}
+        label={i18n.MESSAGE_FIELD_LABEL}
+        isInvalid={isMessageInvalid}
+      >
+        <TextFieldWithMessageVariables
+          index={index}
+          editAction={editSubAction}
+          messageVariables={messageVariables}
+          paramsProperty={'message'}
+          inputTargetValue={subActionParams?.message}
+          errors={errors['subActionParams.message'] as string[]}
+        />
+      </EuiFormRow>
+      <TextAreaWithMessageVariables
+        index={index}
+        editAction={editSubAction}
+        messageVariables={messageVariables}
+        paramsProperty={'description'}
+        inputTargetValue={subActionParams?.description}
+        label={i18n.DESCRIPTION_FIELD_LABEL}
+      />
+      <EuiFormRow data-test-subj="opsgenie-alias-row" fullWidth label={i18n.ALIAS_FIELD_LABEL}>
+        <TextFieldWithMessageVariables
+          index={index}
+          editAction={editSubAction}
+          messageVariables={messageVariables}
+          paramsProperty={'alias'}
+          inputTargetValue={subActionParams?.alias}
+        />
+      </EuiFormRow>
+    </>
+  );
+};
+
+CreateAlertComponent.displayName = 'CreateAlertComponent';
+
+const CloseAlertComponent: React.FC<SubActionProps<OpsgenieCloseAlertParams>> = ({
+  editSubAction,
+  errors,
+  index,
+  messageVariables,
+  subActionParams,
+}) => {
+  const isAliasInvalid =
+    errors['subActionParams.alias'] !== undefined &&
+    errors['subActionParams.alias'].length > 0 &&
+    subActionParams?.alias !== undefined;
+
+  return (
+    <>
+      <EuiFormRow
+        data-test-subj="opsgenie-alias-row"
+        fullWidth
+        error={errors['subActionParams.alias']}
+        isInvalid={isAliasInvalid}
+        label={i18n.ALIAS_FIELD_LABEL}
+      >
+        <TextFieldWithMessageVariables
+          index={index}
+          editAction={editSubAction}
+          messageVariables={messageVariables}
+          paramsProperty={'alias'}
+          inputTargetValue={subActionParams?.alias}
+          errors={errors['subActionParams.alias'] as string[]}
+        />
+      </EuiFormRow>
+      <TextAreaWithMessageVariables
+        index={index}
+        editAction={editSubAction}
+        messageVariables={messageVariables}
+        paramsProperty={'note'}
+        inputTargetValue={subActionParams?.note}
+        label={i18n.NOTE_FIELD_LABEL}
+      />
+    </>
+  );
+};
+
+CloseAlertComponent.displayName = 'CloseAlertComponent';
+
+const OpsgenieParamFields: React.FC<ActionParamsProps<OpsgenieActionParams>> = ({
+  actionParams,
+  editAction,
+  errors,
+  index,
+  messageVariables,
+}) => {
+  const { subAction, subActionParams } = actionParams;
+
+  const currentSubAction = useRef<string>(subAction ?? OpsgenieSubActions.CreateAlert);
+
+  const actionOptions = [
```

Comment:
nit: this can be moved outside the component as it is not depend on the props or the state.